src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +174,313 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
   }
 }
 
+TEST_F(SetAppIconRequestTest, OnEvent_InvalidEnum_ReceivedUnknownEvent) {
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  Event event(hmi_apis::FunctionID::INVALID_ENUM);
+  event.set_smart_object(*msg);
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+  request_->on_event(event);
+}
+
+TEST_F(SetAppIconRequestTest, OnEvent_NonExistentApp_NullPointer) {
+  MockAppPtr mock_app_nonexistent;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_nonexistent));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::UNSUPPORTED_RESOURCE;
+  (*msg)[am::strings::msg_params][am::strings::info] = "info1";
+
+  Event event(hmi_apis::FunctionID::UI_SetAppIcon);
+  event.set_smart_object(*msg);
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+  request_->on_event(event);
+}
+
+TEST_F(SetAppIconRequestTest, Run_AllCorrectData_SendHMIRequest) {
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_))
```

Comment:
@BSolonenko Is it possible to use: 
```
.Times(2)
      .WillRepeatedly(Return(mock_app_)));
```
???

Reply:
@AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/1974/commits/b69062aa4d5eeea9c38a0ac0745d1afe7bfce283